salt/cli/ssh.py

```diff
@@ -16,6 +16,17 @@ def run(self):
         if '-H' in sys.argv or '--hosts' in sys.argv:
             sys.argv += ['x', 'x']  # Hack: pass a mandatory two options
                                     # that won't be used anyways with -H or --hosts
+        if '--bootstrap' == sys.argv[1]:
+            sys.argv.remove('--bootstrap')
+            sys.argv += ['-r', '\
+                if grep -q "SUSE" /etc/os-release ; then\
+                    zypper -n in python python-xml ;\
+                elif [ -e /etc/debian_version ] ; then\
+                    apt --yes install python ;\
+                else\
+                    echo "please add support for your OS to salt-ssh bootstrap" ;\
+                    exit 6 ;\
+                fi']
```

Comment:
I hate to say that, but no. In the [corresponding issue](https://github.com/saltstack/salt/issues/48224#issuecomment-400281487) I already explained that this will turn into a trimmed, poor's man `pkg.install` remake that will work only for that OS and probably only for that version.

That said, please write your own bootstrap script in whatever scripting language you are about: PowerShell for Windows, Korn Shell for Solaris, Bourne Again Shell for Linux, Shell for BSD etc.

![stop](http://static.tumblr.com/0756fba90f7959b934779228ac049ca5/4zeleik/WSrobcq2x/tumblr_static_62jcuiuvxy4gwcwsokg8c4oo8.png)

Reply:
But then I'd have to get that bootstrap script onto a node first. Can salt-ssh copy files via raw mode, too?
Because state.apply wont work before python is installed.

Reply:
Well, nothing will work at all, if no Python. I would just think it differently: build a script that would make sure your target nodes are requirements compliant. Say, you have 100 machines that needs to be operated with Salt SSH, so `cat list-of-machines` and `bootstrap &` with the script below I gave an example of.